### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by pabra**
add github actions and more strict eslint rules

### CODE REVIEW

**Comment by pabra** [comment]
see the actions [in action](https://github.com/pabra/simple-runtypes/actions?query=workflow%3Aunit-tests)

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
```
Comment (hoeck) [inline .github/workflows/unit-tests.yml:37]: Shouldn't that use the `yarn cache dir`?

File: test/all.test.ts [hunk 2]
```diff
@@ -324,6 +324,7 @@ describe('string', () => {
 
   it('trims strings', () => {
     const rt = sr.string({ trim: true })
+    type T = ReturnType<typeof rt>
```
[the remark this answers is not in the document]
Comment (pabra) [reply test/all.test.ts:327]: Yes, I totally missed that one.

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (pabra) [reply .github/workflows/unit-tests.yml:37]: Also true.

I'll update the PR.

**Comment by pabra** [comment]
badge preview:
unit-tests

**Comment by hoeck** [comment]
Great, thanks for contributing :1st_place_medal:

**Pull request merged**